majifix311/src/main/java/com/example/majifix311/ui/auth/SigninActivity.java

```diff
@@ -0,0 +1,422 @@
+package com.example.majifix311.ui.auth;
+
+import android.app.ProgressDialog;
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.design.widget.Snackbar;
+import android.support.design.widget.TextInputEditText;
+import android.support.design.widget.TextInputLayout;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.text.TextUtils;
+import android.util.Patterns;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+
+import com.example.majifix311.BuildConfig;
+import com.example.majifix311.R;
+import com.example.majifix311.auth.Auth;
+import com.example.majifix311.models.Party;
+import com.jakewharton.rxbinding2.view.RxView;
+import com.jakewharton.rxbinding2.widget.RxTextView;
+
+import java.net.UnknownHostException;
+import java.util.concurrent.TimeUnit;
+
+import io.reactivex.Observable;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.functions.Consumer;
+import io.reactivex.functions.Predicate;
+import io.reactivex.observers.DisposableObserver;
+import io.reactivex.schedulers.Schedulers;
+import retrofit2.adapter.rxjava2.HttpException;
+
+/**
+ * Reactive Signin activity
+ *
+ * @see <a href="https://academy.realm.io/posts/donn-felker-reactive-android-ui-programming-with-rxbinding/">https://academy.realm.io/posts/donn-felker-reactive-android-ui-programming-with-rxbinding/</a>
+ * @see <a href="https://medium.com/@etiennelawlor/rxjava-on-the-sign-in-screen-9ecb66b88572">https://medium.com/@etiennelawlor/rxjava-on-the-sign-in-screen-9ecb66b88572</a>
+ */
+public class SigninActivity extends AppCompatActivity {
+
+    public static final String TAG = SigninActivity.class.getSimpleName();
+
+    //view references
+    private TextInputLayout emailInputLayout;
+    private TextInputLayout passwordInputLayout;
+    private TextInputEditText emailTextInput;
+    private TextInputEditText passwordTextInput;
+    private Button signInButton;
+
+    //rxbindings
+    private CompositeDisposable compositeDisposable;
+    private Observable<CharSequence> emailObservable;
+    private Observable<CharSequence> passwordObservable;
+    private Observable<Object> signInButtonObservable;
```

Comment:
I'm also used to seeing `m` prefix for member variables.

Reply:
Yes it used mostly in Google Code Guidelines for Android. My proplem comes when you have `protected` and `private` member...any differentiation?. Also i oppose it due to code readability. It easy to pronounce `emailObservable` than `mEmailObservable`. Same applies to other classes. 

But am open to discussions, the only problem am having is I normally want everyone to read my code and understand them.

Lets pass over [https://github.com/ribot/android-guidelines/blob/master/project_and_code_guidelines.md](https://github.com/ribot/android-guidelines/blob/master/project_and_code_guidelines.md),
 [https://google.github.io/styleguide/javaguide.html](https://google.github.io/styleguide/javaguide.html),